### PR TITLE
feat: miragejs (#134)

### PR DESCRIPTION
**Pull request by rayacers**
Add mock server. Add tests for breadcrumbs.

### CODE REVIEW

File: cypress/e2e/navigator.cy.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  it('Navigate shared items', () => {
+    // The testing file structure is in the following
+    // - Shared Items
+    //  - folder3
+    //    - sharedFolder1
+    //      - sharedDocument1
+    const FOLDER3_ID = '1111-kkkk';
+    const SHARED_FOLDER1_ID = '1234-kkkk';
+    const SHARED_DOCUMENT1_ID = '5678-kkkk';
```
Comment (pyphilia) [inline cypress/e2e/navigator.cy.ts:108]: Same comment as above.

File: cypress/e2e/navigator.cy.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    // Navigate to the shared folder1 by the dropdown menu
+    cy.visit('/shared');
```
Comment (pyphilia) [inline cypress/e2e/navigator.cy.ts:111]: Use string from the paths file.

File: cypress/support/commands.ts [hunk 3]
```diff
@@ -34,4 +34,24 @@
 //       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
 //     }
 //   }
-// }
\ No newline at end of file
+// }
```
Comment (pyphilia) [inline cypress/support/commands.ts:37]: You can remove the above code.

File: src/components/navigator/Navigator.tsx [hunk 4]
```diff
@@ -107,11 +111,13 @@ const Navigator = () => {
     pathname !== SHARED_ITEMS_PATH &&
     pathname !== HOME_PATH
   ) {
-    return <Navigate to={HOME_PATH} />;
+    return null;
+    // return <Navigate to={HOME_PATH} />;
```
Comment (pyphilia) [inline src/components/navigator/Navigator.tsx:115]: Remove?

File: src/config/selectors.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+export const ROOT_MENU_ID = 'rootMenu';
+export const ROOT_MENU_DROPDOWN_BUTTON_ID = 'rootMenuDropdownButton';
+
+export const buildBreadcrumbsItemLink = (id: string): string =>
+  `breadcrumbsItemLink-${id}`;
+export const buildMenu = (id: string): string => `menu-${id}`;
+export const buildMenuDropdownButton = (id: string): string =>
+  `menuDropdownButton-${id}`;
+export const buildMenuItem = (id: string, parent: string): string =>
+  `menuitem-${parent}-${id}`;
```
Comment (pyphilia) [inline src/config/selectors.ts:16]:
```suggestion
`menuItem-${parent}-${id}`;
```

File: src/mockServer/mockData/items.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import { Item } from '@graasp/sdk';
+
+const MOCK_ITEMS: Item[] = [
+  {
+    id: '1111-bbbb',
+    name: 'folder1',
+    path: '1111_bbbb',
```
Comment (pyphilia) [inline src/mockServer/mockData/items.ts:7]: Let's have correct id and path for all your items.

File: src/mockServer/mockData/members.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { Member, MemberType } from '@graasp/sdk';
+
+const MOCK_MEMBERS: Member[] = [
+  {
+    id: 'user1',
```
Comment (pyphilia) [inline src/mockServer/mockData/members.ts:5]: Same comment as above.

File: src/mockServer/mockData/membership.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { ItemMembership, PermissionLevel } from '@graasp/sdk';
+
+const MOCK_MEMBERSHIP: ItemMembership[] = [
+  {
+    id: 'membership1',
```
Comment (pyphilia) [inline src/mockServer/mockData/membership.ts:5]: Same comment as above.

File: src/mockServer/mockData/membership.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+    memberId: 'user1',
+    itemPath: '1111_kkkk.1234_kkkk',
```
Comment (pyphilia) [inline src/mockServer/mockData/membership.ts:7]: Would it be possible to refer to the mock data directly?

File: src/mockServer/mockData/membership.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import { ItemMembership, PermissionLevel } from '@graasp/sdk';
+
+const MOCK_MEMBERSHIP: ItemMembership[] = [
+  {
+    id: 'membership1',
+    memberId: 'user1',
```
Comment (pyphilia) [inline src/mockServer/mockData/membership.ts:6]: Would it be possible to refer to the mock data directly?

File: src/mockServer/mockData/membership.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    itemPath: '1111_kkkk.1234_kkkk',
+    permission: PermissionLevel.Admin,
+    createdAt: Date.now().toLocaleString(),
+    updatedAt: Date.now().toLocaleString(),
+    creator: 'user2',
```
Comment (pyphilia) [inline src/mockServer/mockData/membership.ts:11]: Would it be possible to refer to the mock data directly?

File: src/mockServer/route.ts [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+import { UUID } from '@graasp/sdk';
+
+export const ITEMS_ROUTE = 'items';
+export const buildGetItemRoute = (id: UUID): string => `${ITEMS_ROUTE}/${id}`;
+export const buildGetChildrenRoute = (id: UUID, ordered: boolean): string =>
+  `${ITEMS_ROUTE}/${id}/children${qs.stringify(
+    { ordered },
+    { addQueryPrefix: true },
+  )}`;
```
Comment (pyphilia) [inline src/mockServer/route.ts:11]: You can get these from query-client from API_ROUTES I think. If you can't then just strings as the other endpoints for consistency.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks a lot for this PR, we're aiming to add this mock server to all interfaces but never have the time, this is a very nice first step.
Let me know if you have questions.

File: src/components/navigator/Navigator.tsx [hunk 13]
```diff
@@ -107,11 +111,12 @@ const Navigator = () => {
     pathname !== SHARED_ITEMS_PATH &&
     pathname !== HOME_PATH
   ) {
-    return <Navigate to={HOME_PATH} />;
+    return null;
```
Comment (pyphilia) [inline src/components/navigator/Navigator.tsx:114]: You don't redirect anymore?

File: src/mockServer/types.ts [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+export type Member = {
+  id: UUID;
+  name: string;
+  email: string;
+  extra: MemberExtra;
+  createdAt: string;
+  updatedAt: string;
+};
```
Comment (pyphilia) [inline src/mockServer/types.ts:42]: Delete this file. Don't define your own types. You can get them all from `@graasp/sdk`

**Review by pyphilia — APPROVED** [review]
Last comments and you can merge! Great job 🌹

File: src/components/navigator/Navigator.tsx (hunk 13 above)
Comment (rayacers) [reply src/components/navigator/Navigator.tsx:114]: No. I currently return an empty content.

**Pull request merged**